lib/confluence/metric_computer.es6.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  name: 'MetricComputer',
+  package: 'org.chromium.apis.web',
+  documentation: `MetricComputer is a class that process all browsers
+      with different vendors and call compute methods on each date
+      when there is a new release of any vendors' browser.`,
+  requires: [
+    'org.chromium.apis.web.Browser',
+    'org.chromium.apis.web.WebInterface',
+    'org.chromium.apis.web.BrowserWebInterfaceJunction',
+    'foam.mlang.ExpressionsSingleton',
+    'foam.dao.ArraySink',
+    'foam.dao.EasyDAO',
+  ],
+  properties: [
+    {
+      name: 'browserApiDAO',
+      documentation: `A DAO containing junction objects of Browser and
+        WebInterface.`,
+      typeName: 'BrowserAPI DAO',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'browserDAO',
+      documentation: `A DAO containing all browser infomation.`,
+      typeName: 'Browser DAO',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'interfaceDAO',
+      documentation: `A DAO containing all interface and API pairs.`,
+      typeName: 'WebInterface DAO',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'mlang',
+      documentation: `Object contains mlang expressions, it supports
+        logic operations such as AND and OR and math operations such
+        as IN, EQ.`,
+      factory: function() {
+        return this.ExpressionsSingleton.create();
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'init',
```

Comment:
Also, the complexity here, I think, warrants some tests against a reduced data set to convince ourselves the logic is right. Moving the bulk of the code out of `init()` into a method that isn't automatically called will ease testing.